### DISCUSSION TITLE
What is uapi and nlas?

### DISCUSSION THREAD

**Discussion by 0x241F31** [DefGuard/wireguard-rs#49] [Q&A] [answered]
Thank you for open sourcing this library. I'm wondering what does uapi/nlas mean and when should I use it?

**Comment by moubctez** [comment, marked as answer]
UAPI is the WireGuard API for UNIX socket communication with WireGuard implementation that uses it (e.g. wireguard-go).

NLA is NetLink Attribute. NetLink is a kernel interface for networking resources (https://www.kernel.org/doc/html/next/userspace-api/netlink/intro.html). WireGuard kernel module on Linux uses NLAs for configuration.
